### PR TITLE
Mepts 908  | list of patients elegible for vl

### PR DESCRIPTION
**Pull request by eliazmuchave**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java [hunk 1]
```diff
@@ -0,0 +1,1166 @@
...
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   *
+   * <p>X1- select all patients with “Data da próxima consulta” (concept id 1410, value_datetime) as
+   * “Last Next scheduled Consultation Date” from the most recent FICHA CLÍNICA (encounter type 6,9)
+   * by report end date (encounter_datetime <= endDate) and “Last Next scheduled Consultation Date”
+   * >= startDate and <= endDate
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java:113]: Ni @eliazmuchave, I dont see this section in the query

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
Please have a look

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java (hunk 1 above)
Comment (eliazmuchave) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java:113]: @psmatsinhe, I fixed it

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java [hunk 2]
```diff
@@ -0,0 +1,1167 @@
...
+    valuesMap.put("23866", hivMetadata.getArtDatePickupMasterCard().getConceptId());
+    valuesMap.put("23865", hivMetadata.getArtPickupConcept().getConceptId());
+    valuesMap.put("1065", hivMetadata.getPatientFoundYesConcept().getConceptId());
+
+    String query =
+        " SELECT p.patient_id FROM patient p "
+            + "INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + "INNER JOIN "
+            + "(SELECT p.patient_id, MAX(o.value_datetime) AS last_obs, o.obs_datetime "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java]: @eliazmuchave this , o.obs_datetime will not git you the right result

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java [hunk 3]
```diff
@@ -0,0 +1,1167 @@
...
+            + "WHERE p.voided = 0 "
+            + "  AND e.encounter_type = ${52} "
+            + "  AND e.location_id = :location "
+            + "  AND o.concept_id = ${23865} "
+            + "  AND o.value_coded = ${1065} "
+            + "  AND o.obs_datetime = most_recent.obs_datetime "
+            + "  AND o.voided = 0 "
+            + "  AND e.voided = 0 "
+            + "  AND p.voided = 0 "
+            + "  AND DATE_ADD(most_recent.last_obs, INTERVAL 30 DAY) BETWEEN :startDate AND :endDate "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java]: Review this

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java [hunk 4]
```diff
@@ -0,0 +1,1170 @@
...
+            + "                     AND o.voided = 0  "
+            + "                     AND o.concept_id = ${5096}  "
+            + " AND o.value_datetime BETWEEN :startDate AND :endDate  "
+            + " GROUP BY p.patient_id  "
+            + " UNION  "
+            + " SELECT p.patient_id, most_recent.last_obs AS recent_date FROM patient p  "
+            + " INNER JOIN encounter e ON e.patient_id = p.patient_id  "
+            + " INNER JOIN obs o ON o.encounter_id = e.encounter_id  "
+            + " INNER JOIN  "
+            + " (SELECT p.patient_id, MAX(o.value_datetime) AS last_obs, o.obs_datetime  "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java:407]: @eliazmuchave we already discussed this! Have a look

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java [hunk 5]
```diff
@@ -0,0 +1,1170 @@
...
+   * Pediatria (encounter type 6,9, encounter_datetime) or Ficha Clinica (encounter type 6,
+   * encounter_datetime) or Ficha Resumo (encounter type 53, obs_datetime ) or FSR form (encounter
+   * type 51, encounter_datetime) by start of reporting period (<=startDate) and the Result is >=
+   * 1000 copias/ml (concept 856 value_numeric >= 1000)
+   *
+   * </blockquote>
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getPatientsWithRecentVLIgualOrGreaterThan1000() {
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java]: @eliazmuchave can you update this to be generic as the name of this function getPatientsWithVLLessThan1000()

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java (hunk 5 above)
Comment (eliazmuchave) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java]: I'm not clear

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**